### ISSUE TITLE
Touch interaction fails if touching multiple maps at same time

### ISSUE THREAD

**Issue by navenedrob** [mapbox/mapbox-gl-js#6446]
**mapbox-gl-js version**: 0.44.1

### Steps to Trigger Behavior

https://jsfiddle.net/mth91raa/7/

1. Open example on a multi-touch capable device (phone, touchscreen...)
2. Start a touch drag on one map using left hand.
3. Start a touch drag on a second map using right hand.
4. First map stops working while second map takes over.

### Expected Behavior

Maps can be touched and used independently regardless of what map is being touched or how many maps are available to be touched.

### Actual Behavior

If touching a map and a second map is touched at the same time the first map is no longer able to be interacted with.

**Comment by jfirebaugh** [comment]
Duplicate of #5504.

**Comment by navenedrob** [comment]
@jfirebaugh Can we close the other one and lieu of this one (I forgot about this other one). This one has the fiddle.

**Comment by jfirebaugh** [comment]
Sure.

**Comment by jfirebaugh** [comment]
The reason this happens is that the drag pan handler captures move events at the `document` level. Without support for pointer events and [setPointerCapture](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture) (which are not supported by WebKit), this is the only way to continue the pan with a gesture that goes outside the bounds of the map container.

**Comment by navenedrob** [comment]
I did see that in the source. For my project, I'm OK with limiting the user interaction to the container instead of having it extend beyond the container. In this case, would you just recommend I switch the element that the listener is attached to?

**Comment by jfirebaugh** [comment]
Yeah, see if replacing `window.document` with `this._el` in `drag_pan.js` and `touch_zoom_rotate.js` works.

**Comment by navenedrob** [comment]
I didn't like the idea of modifying the source so I created a solution for myself without having to do that. Out of the box this issue will still be present.

I disabled interaction by just setting the interactive option to false then wrote a little pointer events class and borrowed the drag and pan code (from 0.44.2) integrating it with equivalent pointer events (each is a method in my class to keep things tidy). Then, I just added my pointer event listeners to the canvas container. Works like a charm and now I can drag and pan on any map independently. My use case allows me to be able to use pointer events, so I did.

An alternative to adding it to the canvas container directly is positioning the controls container absolutely over top of the canvas container and adding the listeners to that instead. With this method you would need to manually dispatch a touch event to the canvas container.

I could add pinch zoom but in my use case I do not have that need.
